docs/canary_releases.md

```diff
@@ -0,0 +1,70 @@
+# Canary releases
+
+There is a need to use canary releases in order to make sure newer versions
+of the gathering conditions don't break the Insights Operator gatherer.
+
+This is how a canary release would look like:
+
+```mermaid
+sequenceDiagram
+    actor RT as Rules Team
+    actor PT as Processing Team
+    participant GC as Gathering Conditions
+    participant AI as App Interface
+    participant GCS as Gathering Conditions Service
+    participant Unleash
+    participant Storage
+    actor C as Cluster
+    actor CC as Canary Cluster
+
+    Note left of GCS: Latest version is 1.0.1
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Get version 1.0.1
+
+    RT ->> GC: Push a new tag 1.0.2
+    RT ->> AI: Set the canary and stable versions<br/>CANARY_VERSION = 1.0.2<br/>STABLE_VERSION = 1.0.1
+    PT ->> AI: Review and merge the PR
+    AI ->> GCS: Update prod deployment
+    
+    rect rgb(250,128,114)
+        alt canary or stable version not found
+            GCS ->> GCS: CrashLoopBackOff
+            PT ->> AI: Revert the PR
+        end
+    end
+
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Still get version 1.0.1
+
+    RT ->> Unleash: Increase the canary population to X%
+    GCS -->> C: Get version 1.0.1
+    rect rgb(143,188,143)
+        GCS -->> CC: Get version 1.0.2
+    end
+    
+    RT ->> Storage: Analyze the canary archives and check they are generated well
+
+    alt canary archives are wrong
+        RT ->> AI: Revert the PR
+        AI ->> GCS: Update prod deployment
+        GCS -->> C: Get version 1.0.1
+        GCS -->> CC: Get version 1.0.1
+    else canary archives look fine
+        RT ->> AI: Remove the canary version and set stable <br/>CANARY_VERSION = none<br/>STABLE_VERSION = 1.0.2
```

Comment:
I suppose this would be another PR. Could you describe this part at the same level of granularity as the first change, please?

Reply:
I added `PT ->> AI: Review and merge the PR`. Not sure what else to say. There are no more steps here. Once it's merged, it should be automatically deployed to prod.

Reply:
That's OK. RT and PT are two separate actors. The diagram should capture whenever either one needs to take an action.